src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -220,25 +188,14 @@ public StudentAttributes getStudentForRegistrationKey(String registrationKey) {
             // CourseStudent
             String originalKey = StringHelper.decrypt(registrationKey.trim());
             CourseStudent courseStudent = getCourseStudentEntityForRegistrationKey(originalKey);
-            if (courseStudent != null) {
-                return new StudentAttributes(courseStudent);
+            if (courseStudent == null) {
+                return null;
             }
+            return new StudentAttributes(courseStudent);
         } catch (Exception e) {
             // even if the student does not exist, it's not supposed to throw an exception
-            log.severe("Exception thrown trying to retrieve CourseStudent \n"
+            Assumption.fail("Exception thrown trying to retrieve CourseStudent \n"
```

Comment:
@wkurniawan07, I am having problems here. In the old code with Student, this used to just return null. There's a test in `StudentFeedbackResultsPageUiTest` where a registration key looking like `...${amp}next=/page/studentFeedbackResultsPage?courseid=SFResultsUiT.CS2104${amp}studentemail=drop.out@gmail.tmt${amp}fsname=First Session${amp}key=...` was used as a registration key. I think that url pattern was a workaround done a while back due to Google dropping parameters when they redirect the user. I'll silence this error here for the time being, but I think something is wrong with that registration key handling (the 'desanitization' should have happened before passing that regkey into this method?).